internal/paascharm/charmcraft_test.go

```diff
@@ -0,0 +1,103 @@
+package paascharm_test
+
+import (
+	"testing"
+
+	"github.com/kr/pretty"
+
+	"github.com/canonical/paascharmgen/internal/paascharm"
+)
+
+func TestGenerateGoStructsFromCharmcraftYaml(t *testing.T) {
+	yamlData := `
+name: go-app
+type: charm
```

Comment:
Is there any way we could dynamically generate this using charmcraft? Just so that we know that the current output of charmcraft works?

Reply:
I have changed one of the spread tests to use `charmcraft.yaml` generated with `charmcraft init --profile go-framework`.

Reply:
(reopened for discussion)